### PR TITLE
(650a) Add support for updating the status of a referral

### PR DESCRIPTION
**Pull request by jsrobertson**
## Context

In anticipation of the referral complete step.

## Changes in this PR
- Add API path for updating status
- Add client and service methods for updating a referral status using the endpoint

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: server/@types/models/Referral.ts [hunk 1]
```diff
@@ -18,4 +18,6 @@ type ReferralUpdate = {
   reason?: string
 }
 
-export type { CreatedReferralResponse, Referral, ReferralUpdate }
+type ReferralUpdateStatus = 'awaiting_assesment' | 'assessment_started' | 'referral_started' | 'referral_submitted'
```
Comment (yndajas) [inline server/@types/models/Referral.ts:21]: Is this not just a `ReferralStatus`? I think we should be able to reuse this later when we need to read the status?

**Review by yndajas — APPROVED** [review]
Looks good. Just one suggestion to make the type more generic/reusable

File: server/@types/models/Referral.ts (hunk 1 above)
Comment (jsrobertson) [reply server/@types/models/Referral.ts]: Good shout!

**Review by Gweaton — APPROVED** [review]
Looks good to me - looks like the tests are failing after renaming to `ReferralStatus` but not updating the import in `referralService.test.ts` but this is good to go after fixing that.

**Pull request merged**